dbms/src/Flash/Coprocessor/DAGContext.h

```diff
@@ -197,14 +204,28 @@ class DAGContext
             warnings_.push_back(error);
         }
     }
+    void fillWarnings(tipb::SelectResponse & response)
+    {
+        std::vector<tipb::Error> warnings_vec;
+        consumeWarnings(warnings_vec);
+        for (auto & warning : warnings_vec)
+        {
+            auto * warn = response.add_warnings();
+            // TODO: consider using allocated warnings to prevent copy?
+            warn->CopyFrom(warning);
+        }
+        response.set_warning_count(getWarningCount());
+    }
     void clearWarnings()
     {
         warnings.clear();
         warning_count = 0;
     }
     UInt64 getWarningCount() { return warning_count; }
     const mpp::TaskMeta & getMPPTaskMeta() const { return mpp_task_meta; }
-    bool isBatchCop() const { return is_batch_cop; }
+    bool isCop() const { return !is_mpp_task && cop_kind == CoprocessorKind::Cop; }
+    bool isCopStream() const { return !is_mpp_task && cop_kind == CoprocessorKind::CopStream; }
+    bool isBatchCop() const { return !is_mpp_task && cop_kind == CoprocessorKind::BatchCop; }
```

Comment:
Maybe we can rename `CoprocesorKind` to `RequestKind`, then it can include mpp task: `Cop/CopStream/BatchCop/MPP`